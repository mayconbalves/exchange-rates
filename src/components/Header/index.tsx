import styles from './styles.module.css'

const Header = () => {
  return (
    <header className={styles.wrapper}>
      <img src="./logo.png" alt="B3 logo" />
      <h2 className={styles.wrapper__title}>Exchange Rates</h2>
    </header>
  )
}

export default Header
