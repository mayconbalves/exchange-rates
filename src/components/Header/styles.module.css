.wrapper {
  align-items: center;
  background-color: #001b74;
  display: flex;
  height: 72px;
  width: 100%;
}

.wrapper__title {
  color: #ffffff;
  font-size: large;
  margin: 5px;
}
