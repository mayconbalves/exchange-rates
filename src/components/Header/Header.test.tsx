import { render, screen } from '@testing-library/react'
import Header from '.'

describe('Header Component', () => {
  it('Should render the header element', () => {
    render(<Header />)
    const headerElement = screen.getByRole('banner')
    expect(headerElement).toBeInTheDocument()
  })

  it('Should display the logo image with correct alt text', () => {
    render(<Header />)
    const logoImage = screen.getByRole('img', { name: /b3 logo/i })
    expect(logoImage).toBeInTheDocument()
    expect(logoImage).toHaveAttribute('src', './logo.png')
  })

  it('Should display the title text "Exchange Rates"', () => {
    render(<Header />)
    const titleElement = screen.getByText(/exchange rates/i)
    expect(titleElement).toBeInTheDocument()
  })

  it('Should be create snapshot', () => {
    const { container } = render(<Header />)

    expect(container).toMatchSnapshot()
  })
})
