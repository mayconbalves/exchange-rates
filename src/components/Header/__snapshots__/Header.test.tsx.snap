// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should be create snapshot 1`] = `
<div>
  <header
    class="wrapper"
  >
    <img
      alt="B3 logo"
      src="./logo.png"
    />
    <h2
      class="wrapper__title"
    >
      Exchange Rates
    </h2>
  </header>
</div>
`;
